lib/loggly/common.js

```diff
@@ -247,32 +272,42 @@ common.loggly = function () {
     }
 
     if (arrMsg.length === arrSize) {
-      sendBulkLogs();
+      popMsgsAndSend();
     }
   }
   else if(isValidToken) {
-    sendLogs();
+    if (requestBody) {
+      popMsgsAndSend();
+    }
   }
 
   //
-  //function to retry sending logs maximum 5 times if server error occurs
+  //function to retry sending logs maximum 5 times if any error occurs
   //
-  function retryOnServerError(err) {
-    if (!arrRetryLogs.length) return;
-    else {
-      if (notFailedOnServerError && totalRetries >= maxRetryAllowed) {
-        console.log('Failed after ' + totalRetries + ' retries on error - ' + statusCode, '"'+err.statusMessage+'"');
-        notFailedOnServerError = false;
-        arrRetryLogs.length = 0;
-        totalRetries = 0;
+  function retryOnError(mode, response) {
+    function tryAgainIn(sleepTimeMs) {
+      console.log('log #' + mode.id + ' - Trying again in ' + sleepTimeMs + '[ms], attempt no. ' + mode.attemptNumber);
+      setTimeout(function () {
+        isBulk ? sendBulkLogs(mode) : sendInputLogs(mode);
+      }, sleepTimeMs);
+    }
+    if (mode.attemptNumber >= numberOfRetries) {
```

Comment:
@mchaudhary This condition checks if the total number of retries has reached the maximum retrying limit then show the failed event message on console.